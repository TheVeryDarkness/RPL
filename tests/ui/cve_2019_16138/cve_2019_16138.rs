//@compile-flags: -Z inline-mir=false

// #[rpl::dump_mir(dump_cfg, dump_ddg)]
fn foo() {
    let pixel_count = 1920 * 1080;
    let mut ret: Vec<(u8, u8, u8)> = Vec::with_capacity(pixel_count);
    unsafe {
        ret.set_len(pixel_count);
        //~^ERROR: it violates the precondition of `std::vec::Vec::set_len` without initializing its content
    }
}
