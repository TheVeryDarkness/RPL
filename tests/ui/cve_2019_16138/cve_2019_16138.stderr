error: it violates the precondition of `std::vec::Vec::set_len` without initializing its content
  --> tests/ui/cve_2019_16138/cve_2019_16138.rs:8:9
   |
LL |     let mut ret: Vec<(u8, u8, u8)> = Vec::with_capacity(pixel_count);
   |                                      ------------------------------- `std::vec::Vec` created here
LL |     unsafe {
LL |         ret.set_len(pixel_count);
   |         ^^^^^^^^^^^^^^^^^^^^^^^^
   |
   = help: before calling `set_len` to extend its length, make sure all elements are initialized, such as using `spare_capacity_mut` or `as_mut_ptr`

error: aborting due to 1 previous error

